w3-blob.md

```diff
@@ -0,0 +1,419 @@
+        "cmd": "/service/blob/accept",
+        "sub": "did:web:web3.storage",
+        "args": {
+          "space": "did:key:zAlice",
+          "blob": {
+            // multihash of the blob as byte array
+            "content": { "/": { "bytes": "mEi...sfKg" } },
+            "size": 2_097_152,
+          },
+          // This task is blocked on allocation
+          _: { "await/ok": { "/": "bafy...alloc" } }
+        }
+      }
+    ]
+  }
+}
+```
+
+### Add Blob Capability
+
+#### Add Blob Capability Schema
+
+```ts
+type AddBlob = {
+  cmd: "/space/content/add/blob"
+  sub: SpaceDID
+  args: {
+    blob: Blob
+  }
+}
+
+type Blob = {
+  content: Multihash
+  size:    int
+}
+
+type Multihash = bytes
+type SpaceDID = string
+```
+
+#### Blob Content
+
+The `args.blob.content` field MUST be a [multihash] digest of the blob payload bytes. Implementation SHOULD support SHA2-256 algorithm. Implementation MAY in addition support other hashing algorithms.
+
+#### Blob Size
+
+Blob `args.blob.size` field MUST be set to the number of bytes in the blob content.
+
+### Add Blob Receipt
+
+#### Add Blob Receipt Schema
+
+```ts
+// Only operation specific fields are covered the
+// rest are implied
+type AddBlobReceipt = {
+  out: Result<AddBlobOk, AddBlobError>
+  next: [
+    AllocateBlob,
+    AcceptBlob,
+  ]
+}
+
+type AddBlobOk = {
+  claim: {
+    "await/ok": Link<AcceptBlob>
+  }
+}
+
+type AddBlobError = {
+  message: string
+}
+```
+
+#### Add Blob Result
+
+Invocation MUST fail if any of the following is true
+
+1. Provided **sub**ject space is not provisioned with a provider.
+1. Provided `blob.size` is outside of supported range.
+1. Provided `blob.content` is not a valid [multihash].
+1. Provided `blob.content` [multihash] hashing algorithm is not supported.
+
+Invocation MUST succeed if non of the above is true. Success value MUST be an object with a `claim` field set to [await/ok] of the task that produces [location claim].
+
+Task linked from the `claim` of the success value MUST be present in the receipt effects _(`next` field)_.
+
+#### Add Blob Effects
+
+Successful invocation MUST start a workflow consisting of following tasks, that MUST be set in receipt effects (`next` field) in the following order.
+
+1. [Allocate Blob]
+1. [Accept Blob]
+
+## Allocate Blob
+
+Authorized agent MAY invoke `/service/blob/allocate` capability on the [provider] subject to create a memory address where `blob` content can be written via HTTP `PUT` request.
+
+### Allocate Blob Capability
+
+#### Allocate Blob Capability Schema
+
+```ts
+type BlobAllocate = {
+  cmd:  "/service/blob/allocate"
+  sub:  ProviderDID
+  args: {
+    space:  SpaceDID
+    blob:   Blob
+    cause:  Link<AddBlob>
+  }
+}
+```
+
+#### Allocation Space
+
+The `args.space` field MUST be set to the [DID] of the user space where allocation takes place.
+
+#### Allocation Blob
+
+The `args.blob` field MUST be set to the `Blob` the space is allocated for.
```

Comment:
this is content right? Should we keep same name?

Reply:
It references the `Blob` type defined earlier which is `{ content, size }` pair

Reply:
oh I see, ok that is great then 👍🏼